docs/migration/cosmoshub-4-v9-Lambda-upgrade.md

```diff
@@ -0,0 +1,297 @@
+    - Or by using the auto-download functionality (this is not yet recommended)
+
+If you prefer to use Cosmovisor to upgrade, some preparation work is needed before upgrade.
+
+### Method I: Manual Upgrade
+
+Make sure Gaia v9.0.0 is installed by either downloading a [compatable binary](https://github.com/cosmos/gaia/releases/tag/v9.0.0), or building from source. Building from source requires go 1.18.
+
+Run Gaia v8.0.1 till upgrade height, the node will panic:
+
+```shell
+ERR UPGRADE "v9-Lambda" NEEDED at height: 14470501: upgrade to v9-Lambda and applying upgrade "v9-Lambda" at height:14470501
+```
+
+Stop the node, and switch the binary to Gaia v9.0.0 and re-start by `gaiad start`.
+
+It may take several minutes to a few hours until validators with a total sum voting power > 2/3 to complete their node upgrades. After that, the chain can continue to produce blocks.
+
+### Method II: Upgrade using Cosmovisor
+
+> **Warning**  <span style="color:red">**Please Read Before Proceeding**</span><br>
+> **Using Cosmovisor 1.2.0 and higher requires a lowercase naming convention for upgrade version directory. For Cosmovisor 1.1.0 and earlier, the upgrade version is not lowercased.**
+>
+> **For Example:** <br>
+> **Cosmovisor =< `1.1.0`: `/upgrades/v9-Lambda/bin/gaiad`**<br>
+> **Cosmovisor >= `1.2.0`: `/upgrades/v9-lambda/bin/gaiad`**<br>
+
+| Cosmovisor Version | Binary Name in Path |
+|--------------------|---------------------|
+| 1.3                | v9-lambda           |
+| 1.2                | v9-lambda           |
+| 1.1                | v9-Lambda           |
+| 1.0                | v9-Lambda           |
+
+
+
+### _Manually preparing the Gaia v9.0.0 binary_
+
+##### Preparation
+
+Install the latest version of Cosmovisor (`1.3.0`):
+
+```shell
+go install github.com/cosmos/cosmos-sdk/cosmovisor/cmd/cosmovisor@latest
+```
+
+**Verify Cosmovisor Version**
+```shell
+cosmovisor version
+cosmovisor version: v1.3.0
+```
+
+Create a cosmovisor folder:
+
+create a Cosmovisor folder inside `$GAIA_HOME` and move Gaia v8.0.1 into `$GAIA_HOME/cosmovisor/genesis/bin`
+
+```shell
+mkdir -p $GAIA_HOME/cosmovisor/genesis/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/genesis/bin
+````
+
+build Gaia v9.0.0, and move gaiad v9.0.0 to `$GAIA_HOME/cosmovisor/upgrades/v9-lambda/bin`
+
+```shell
+mkdir -p  $GAIA_HOME/cosmovisor/upgrades/v9-lambda/bin
+cp $(which gaiad) $GAIA_HOME/cosmovisor/upgrades/v9-lambda/bin
+```
+
+Then you should get the following structure:
+
+```shell
+.
+├── current -> genesis or upgrades/<name>
+├── genesis
+│   └── bin
+│       └── gaiad  #v8.0.1
+└── upgrades
+    └── v9-lambda
+        └── bin
+            └── gaiad  #v9.0.0
+```
+
+Export the environmental variables:
+
+```shell
+export DAEMON_NAME=gaiad
+# please change to your own gaia home dir
+export DAEMON_HOME= $GAIA_HOME
+export DAEMON_RESTART_AFTER_UPGRADE=true
+```
+
+Start the node:
+
+```shell
+cosmovisor start --x-crisis-skip-assert-invariants
```

Comment:
```suggestion
cosmovisor run --x-crisis-skip-assert-invariants
```

Reply:
i tested locally, here i also have to add --home,  even tough `$DENOM_HOME` is already set